docs/source/user_guide.md

```diff
@@ -153,8 +165,190 @@ name: new-project
 environments:
   default:
     - environment.yml
+variables: {}
+commands: {}
+```
+
+## Environment variables
+
+Conda Project supports defining environment variables in the `conda-project.yml` file that will
+be set upon environment activation or when running commands. Variables are defined in the
+`variables:` key and can have an optional default value. Starting from the minimal example above
+I'll add a variable `FOO` with the default value `has-default-value` and the variable `BAR` with
+no default value.
+
+```yaml
+name: new-project
+environments:
+  default:
+    - environment.yml
+
+variables:
+  FOO: has-default-value
+  BAR:
+
+commands: {}
+```
+
+The values of the variables can be set or overridden by the use of a `.env` file (see the
+[python-dotenv documentation](https://saurabh-kumar.com/python-dotenv/#file-format) for more details) or
+by setting the variable in your shell.
+
+In this example the `BAR` environment variable is unset so `conda project activate` will not start
+unless a value is provided by the shell or in a `.env` file.
+
+```
+❯ conda project activate
+CondaProjectError: The following variables do not have a default value and values
+were not provided in the .env file or set on the command line when executing 'conda project run':
+BAR
+```
+
+On Unix you can do the following
+
+```
+❯ BAR=set-on-cli conda project activate
+## Project environment default activated in a new shell.
+## Exit this shell to de-activate.
+❯ conda activate /Users/adefusco/Development/conda-incubator/conda-project/examples/p/envs/default
+```
+
+On Windows in either `cmd.exe` you can use `set variable=value`
+
+```
+> set BAR=set-in-shell
+> conda project activate
+```
+
+and finally, in Powershell you would use `$env`
+
+```
+> $env:BAR = 'set-in-shell'
+> conda project activate
+```
+
+## Defining and running commands
+
+Conda Project supports running commands as if the desired environment were activated, similar
+to [conda run](https://docs.conda.io/projects/conda/en/latest/commands/run.html).
+
+In Conda Project the `run` command can be used to execute ad-hoc commands or those defined
+in the `conda-project.yml` file. Note that `conda project run` should not be utilized after
+`conda project activate`. The `run` command will not work from within an activated environment.
```

Comment:
What is the failure condition? If there is a nice error message, it might be nice to state it here.

Reply:
As I was writing this I realized I hadn't tested this. Actually, I bet it is safe. 🤔